docs/testing/e2e_test_suite.rst

```diff
@@ -0,0 +1,249 @@
+       - ``Page/``: Stores UI locators for each page. Avoid hard-coding complex CSS or XPath locators in tests; instead, use PageObject classes.
+       - ``Step/``: Contains step objects that group common functionalities for tests.
+   * - ``acceptance/``
+     - Contains acceptance tests.
+
+.. vale off
+
+Writing and Running Tests
+**************************
+
+.. vale on
+
+Writing tests
+=============
+
+Writing tests in Codeception involves creating files within the tests/Acceptance directory. Each file contains a class with methods that define the test scenarios.
+
+1. Create a New Test File: Use the following command to generate a new file:
+
+.. code-block:: bash
+
+    bin/codecept generate:cest acceptance <TestName>
+
+This command will create a file named TestSuiteNameCest.php inside the tests/Acceptance directory.
+
+2. Define Test Scenarios: Open the generated file and define your test scenarios. Each method within the class represents a different scenario. Use Codeception's built-in assertions and helper functions to verify the expected outcomes. Here’s a simple example:
+
+.. code-block:: PHP
+
+    <?php
+
+    class TestSuiteNameCest
+    {
+        public function _before(AcceptanceTester $I)
+        {
+            // Code to run before each test
+        }
+
+        public function _after(AcceptanceTester $I)
+        {
+            // Code to run after each test
+        }
+
+        // Define your test methods
+
+        public function login(AcceptanceTester $I)
+        {
+            $I->amOnPage('/s/login');
+            $I->fillField('#username', $name);
+            $I->fillField('#password', $password);
+            $I->click('button[type=submit]');
+            $I->see('Dashboard');
+        }
+    }
+
+3. Utilize PageObjects and StepObjects:
+
+Organize your tests by using PageObject and StepObject classes. This keeps your tests clean and maintainable by separating locators and test steps into reusable components.
+
+- Generate PageObject with:
+
+.. code-block:: bash
+
+    bin/codecept generate:pageobject acceptance ExamplePage
+
+This will create an ExamplePage.php file in /tests/Support/Page/.
+
+- Generate step objects with:
+
+.. code-block:: bash
+
+    bin/codecept generate:stepobject acceptance Example
+
+This will create an Example.php file in/tests/Support/Step/Acceptance.
+
+Running Tests
+=============
+
+You can start tests using the run command provided by Codeception. Here are different ways to run your tests:
+
+**Run All Tests**
+
+.. code-block:: bash
+
+    bin/codecept run
+
+**Run All Acceptance Tests**
+
+.. code-block:: bash
+
+    bin/codecept run acceptance
+
+**Run a Specific Test File**
+
+If you need to run a specific test file, such as ContactManagementCest, use:
+
+.. code-block:: bash
+
+    bin/codecept run acceptance ContactManagementCest
+
+**Run a Specific Test Scenario**
+
+To execute a specific scenario within a test file, you can specify the test method like this:
```

Comment:
```suggestion
To run a specific scenario within a test file, you can specify the test method like this:
```

Let's be consistent and use 'run' rather than execute.